source/istio_gateway.go

```diff
@@ -172,6 +172,11 @@ func (sc *gatewaySource) Endpoints(ctx context.Context) ([]*endpoint.Endpoint, e
 			return nil, err
 		}
 
+		if len(gwEndpoints) == 0 {
+			log.Debugf("No endpoints could be generated from gateway %s/%s", gateway.Namespace, gateway.Name)
+			continue
```

Comment:
This is not only logging, adding a `continue` will have the effect of skipping lines 180 to 182. Is this desired?

Reply:
That is the intention.

180 reports that something will be added, where actually nothing is as the `gwEndpoints` is empty.

```
Endpoints generated from gateway: example/gateway: []
```

181 and 182 will not do anything as there is nothing in the endpoint array.

So breaking out of the array give a debug message that nothing was found.

Reply:
The `continue` is also present on the [istio_virtualservice.go#L168-L171](https://github.com/kubernetes-sigs/external-dns/blob/master/source/istio_virtualservice.go#L168-L171).

Reply:
@Raffo is there anything else I need to do to get this merged?

Reply:
Yeah that looks good to me.